docs/contributing-guide.adoc

```diff
@@ -1,52 +1,121 @@
-= Contributing guide
+= Contributing to AeroGear JavaScript SDK
 
-== Local setup
+First, ensure you have the https://docs.npmjs.com/[latest npm] installed.
 
-1. Install required dependencies
+Then, to set everything up run in your terminal:
 [source,bash]
-----
 npm install
 npm run bootstrap
-----
-2. Compile code
-[source,bash]
-----
-npm run build
-----
-3. Run tests and lint
+
+== Project Structure
+The SDK is made by a group of modules, each one implemented in a different NPM package and all placed under `/packages`.
+
+Each module provides an API to use a particular AeroGear service.
+
+A main package called `core` contains a shared set of interfaces, classes and other functionalities that the rest of modules depend on.
+
+== Working on modules
+
+Modules are written in https://www.typescriptlang.org/[TypeScript] and tested by https://mochajs.org/[mocha].
+
+Once the development is done, the module is built and the source code is transpiled into a `dist/` folder and published. Typings are automatically generated during this step as well.
+
+== Commands
+
+Run this commands from the repo's root directory.
+
+==== Run tests with coverage
 [source,bash]
-----
 npm run test
-npm run lint
-----
 
-=== Using core in projects
+==== Run linter
+[source,bash]
+npm run lint
 
-1. Link resources
+==== Build
 [source,bash]
-----
-  cd packages/core
-  npm link .
-  npm link @aerogearservices/core
-----
-
-2. Use core in your application
-[source,javascript]
-----
-var config = require('./mobile-config.json')
-var ConfigService = require('@aerogearservices/core').ConfigService;
-
-var config = new ConfigService(config);
-----
-> Note: SDK requires common.js implementation like browserify or webpack to be present
-
-3. Use core config with example SDK
-[source,javascript]
-----
-var keycloakConfig = config.getKeycloakConfig()
-var Keycloak = require('keycloak-js')
-
-var keycloak = Keycloak(keycloakConfig.config);
-// Use keycloak initialized. Use keycloak according to client documentation
-----
+npm run build
+
+== Adding a new module
+
+Create a new NPM package under the packages folder with the module name: `/packages/my-module`.
+
+When creating the package.json, include the org scope to its package name: `"name": "@aerogear/my-module"`.
+
+Then add the following standard options and dependencies:
+[source.json]
+{
```

Comment:
I think it's better to move these big code snippets to separate files, in raincatcher we created a [templates directory](https://github.com/feedhenry-raincatcher/raincatcher-core/tree/master/templates) directory to keep these.

Actually we had full example modules to copy paste into new ones, but since we needed to exclude them from lerna it was hard to use the same scripts to update dependencies, etc.

Still we can start off with moving these base package.json and tsconfig to a new dir.

Reply:
Awesome